Data/Matrix.hs

```diff
@@ -556,6 +556,31 @@ inverse m
             rref'd = rref adjoinedWId
         in rref'd >>= return . submatrix 1 (nrows m) (ncols m + 1) (ncols m * 2)
 
+-- | /O(strassenMixed)/. The inverse of a square matrix.
+--   Inverts blockwise. Significantly faster than naive Gaussian elimination.
+--   Same time complexity as whatever matrix multiplication (*) currently uses.
+--   See https://en.wikipedia.org/wiki/Invertible_matrix#Blockwise_inversion
+blockInverse :: (Fractional a, Eq a) => Matrix a -> Either String (Matrix a)
+blockInverse m
+    | nrows m /= ncols m = inverse m
+    | otherwise = Right $ blockInverseUnsafe m
+
+-- blockInverse that assumes that it is called
+-- on a square matrix.
+blockInverseUnsafe :: (Fractional a, Eq a) => Matrix a -> Matrix a
+blockInverseUnsafe m
+    | nrows m < 10 = either (\_ -> m) id $ inverse m
```

Comment:
It's not any more than it is in the other `inverse` function. I'm working on a drop-in replacement for that function that will close #38 as well. In the meantime, do you say any glaring issues?